content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,556 @@
+contains an efficiently multi-threaded implementation utilized by both
+a local simulator and actual network construction.  Expressions of
+Glacier in Python and Common Lisp are also in limited public review.
+
+
+# Interoperability
+
+There is no current wire protocol for the queries.  Nodes are advised
+to use Waku messages to include their own metadata in serializations
+as needed.  Such a message contains minimally
+
+```n3
+@prefix rdf:         <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs:        <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd:         <http://www.w3.org/2001/XMLSchema#> .
+
+@prefix glacier      <https://rdf.logos.co/protocol/glacier#> .
+
+glacier:query
+  :holds (
+    :_0 [ rdfs:label "round";
+          rdfs:comment "The local number of rounds participating"
+          a xsd:postitiveInteger; ],
+    :_1 [ rdfs:label "id";
+          rdfs:comment "A unique content-addressable URI for the proposal"; # Use DID??
+          a xsd:anyURI ],
+    :_2 [ rdfs:label "opinion";
+          rdfs:comment "The opinion on the proposal",
+          # TODO constrain as an enumeration on three values
+          a xsd:string ] 
+    ) .
+```
+
+## Syntax
+
+The message exchanged are a simple enumeration of three values is not
+currently analyzed reflecting the opinion on the given proposal:
+
+    { YES, NO, UNDECIDED }
+
+when represented via integers, such as choosing 
+ 
+     { -1, +1, 0 }
+
+parity summations across network invariants often become easier to
+represent.
+
+# Sovereignty Considerations
+
+## Privacy
+
+In practice, each honest node gossips its current opinion which
+reduces the number of messages that need to be gossiped for a given
+proposal.  The resulting impact on the privacy of the node's opinion
+is not currently analyzed.
+
+## Security with respect to various Adversarial Models 
+
+Adversarial models have been tested for which the values for current
+parameters of Glacier have been tuned.  Exposition of the
+justification of this tuning need to be completed.
+
+### Local Strategies
+
+#### Random Adversary
+
+A random adversary optionally chooses to respond to all queries with a
+random decision.  Note that this adversary may be in some sense
+Byzantine but not malicious.  The random adversary also models some
+software defects involved in not "understanding" how to derive a truth
+value for a given proposition.
+
+#### Infantile Adversary
+
+Like a petulant child, an infantile adversary responds with the
+opposite vote of the honest majority on an opinion.
+
+### Omniscient Coordinated Behavior Adversaries
+
+An omniscient adversary controls $f$ of $N$ nodes and may inspect,
+delay, and drop arbitrary messages in the gossip layer, and utilize
+this to corrupt consensus away from honest decisions to ones favorable
+to itself.  It may, of course, choose to participate in an honest
+manner until defecting is most advantageous.
```

Comment:
One thing missing here is that we consider the Omniscient adversary to be fully aware at all times of the network state. Meaning that he can always chose to vote in the most efficient way to block the network from reaching finality.

Reply:
TODO Need to start dividing the Omniscient Adversary along complexity classes and coordination power.